packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -175,18 +193,30 @@ export const PremiumContentPurchaseModal = () => {
   if (track && !isValidTrack) {
     console.error('PremiumContentPurchaseModal: Track is not purchasable')
   }
+  const mobile = isMobile()
 
   return (
     <ModalDrawer
       isOpen={isOpen}
       onClose={handleClose}
       onClosed={handleClosed}
-      bodyClassName={styles.modal}
+      bodyClassName={cn(styles.modal, isLoading ? styles.loading : null)}
       isFullscreen
       useGradientTitle={false}
       dismissOnClickOutside
+      zIndex={zIndex.PREMIUM_CONTENT_PURCHASE_MODAL}
+      wrapperClassName={mobile ? styles.mobileWrapper : undefined}
     >
-      {isValidTrack ? (
+      {isLoading ? (
+        <Flex
+          alignItems='center'
+          justifyContent='center'
+          h='100%'
+          css={{ flexGrow: 1 }}
+        >
+          <LoadingSpinner className={styles.spinner} />
```

Comment:
is it possible to have the spinner only next to the "use balance" option? Doesn't feel like we need to wait on the balance to load the rest of the modal?

Reply:
+1